### PR TITLE
[MLR-2215] Releasing report E2E tests

### PR DESCRIPTION
**Pull request by oweingrod-coforma**
## Summary
This PR adds Cypress e2e testing for releasing reports:

### Description
Test scenarios:
* I create/fill/submit and release a report with the correct data pre-filled
* I cannot release a submitted and archived report
* I cannot release a non-locked (non-submitted) report.

### Related ticket
MDCT-2215

### How to test
Pull down the repo and run the Cypress tests—or just verify that this PR doesn't fail PR hooks.

### Important updates
Exported the `traverseRoutes` function from the form e2e test so I could use it.

### Author checklist

- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://bit.ly/3zPrxuZ) tests, if necessary
- [x] I have updated the documentation, if necessary

### CODE REVIEW

File: tests/cypress/tests/e2e/mlr/release.feature [hunk 1]
```diff
@@ -0,0 +1,739 @@
...
+
+# Scenario: Admin user archives the program
+#     Given I am logged in as an admin user
+#     And I am on "/mcpar"
+#     And these form elements are filled:
+#         | state | dropdown | District of Columbia |
+
+#     When I click the "Go to Report Dashboard" button
+#     And I click the "Archive" button
+#     Then the program is archived
```
Comment (ntsummers1) [inline tests/cypress/tests/e2e/mlr/release.feature:737]:
```suggestion

```
Can remove this

Comment (oweingrod-coforma) [reply tests/cypress/tests/e2e/mlr/release.feature]: Oops! Fixed.

**Review by ntsummers1 — APPROVED** [review]

**Review by tblackwe — APPROVED** [review]

File: services/app-api/handlers/reports/submit.ts [hunk 2]
```diff
@@ -147,6 +175,10 @@ export const submitReport = handler(async (event, _context) => {
       status: StatusCodes.SUCCESS,
       body: {
         ...newItem,
+        fieldData: { ...fieldData },
```
[the remark this answers is not in the document]
Comment (oweingrod-coforma) [reply services/app-api/handlers/reports/submit.ts:178]: @tblackwe there isn't—i'll change this.

File: tests/cypress/tests/e2e/mlr/form.stepdef.js [hunk 3]
```diff
@@ -29,7 +29,7 @@ When("I submit a new MLR program", () => {
 
 Then("the program is submitted", () => {});
 
-const traverseRoutes = (routes) => {
+export const traverseRoutes = (routes) => {
```
[the remark this answers is not in the document]
Comment (oweingrod-coforma) [reply tests/cypress/tests/e2e/mlr/form.stepdef.js:32]: I agree—this should be refactored. @braxex Any chance we have time for this refactoring task?

File: tests/cypress/tests/e2e/mlr/release.feature [hunk 4]
```diff
@@ -0,0 +1,739 @@
...
+        When I create, fill, and submit a report
+        Then The report is submitted successfully
+        Given I am logged in as an admin user
+        When I release a report
+        Then The report will have the correct content pre-filled
+
+    Scenario: A report cannot be released if it is archived.
+        Given I am logged in as an state user
+        When I create, fill, and submit a report
+        Then The report is submitted successfully
```
[the remark this answers is not in the document]
Comment (oweingrod-coforma) [reply tests/cypress/tests/e2e/mlr/release.feature:14]: Fixed

File: tests/cypress/tests/e2e/mlr/release.stepdef.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  cy.get("button[type=submit]").contains("Save").click();
+
+  //Find our new program and open it
+  cy.findByText(programName)
+    .parent()
+    .find('button:contains("Enter")')
+    .focus()
+    .click();
+
+  //Using the mcpar.json as a guide, traverse all the routes/forms and fill it out dynamically
```
[the remark this answers is not in the document]
Comment (oweingrod-coforma) [reply tests/cypress/tests/e2e/mlr/release.stepdef.js:131]: Fixed

**Pull request merged**